agent-connection/src/main/kotlin/com/luxoft/blockchainlab/corda/hyperledger/indy/IndyParty.kt

```diff
@@ -86,5 +91,44 @@ class IndyParty(private val webSocket: AgentWebSocketClient, val did: String, va
      * @return observable (Single<>) object emitting a single [ProofInfo] upon subscription
      */
     override fun receiveProof() = webSocket.receiveClassObject<ProofInfo>(this)
+
+    /**
+     * Returns observable ([Single]<>) object, emitting JSON-encocoded Tails file by the given tails hash.
+     *
+     * @param tailsHash string-encocoded Tails file
+     *
+     * @return observable ([Single]<>) object emitting [TailsResponse] object
+     */
+    override fun requestTails(tailsHash: String) : Single<TailsResponse> {
+        val result : Single<TailsResponse> = webSocket.receiveClassObject(this)
+        webSocket.sendClassObject(TailsRequest(tailsHash), this)
+        return result
+    }
+
+    private val requestHandlerRef: AtomicReference<(TailsRequest)->TailsResponse> =
+        AtomicReference<(TailsRequest)->TailsResponse> { TailsResponse(it.tailsHash, mapOf())}
+
+    private lateinit var tailRequestMessageHandler: (TailsRequest) -> Unit
```

Comment:
`lateinit` is ambiguous if variable initialization goes inside `init` block

Reply:
ok

Reply:
No it's not. This won't build unless specified.